SonarQube.Scanner.MSBuild.sln

```diff
@@ -1,45 +1,45 @@
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SonarQube.TeamBuild.PreProcessor.Tests", "tests\SonarQube.TeamBuild.PreProcessor.tests\SonarQube.TeamBuild.PreProcessor.Tests.csproj", "{81E8D103-270F-4363-87D1-72BE7FC4F94A}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SonarScanner.MSBuild.PreProcessor.Tests", "tests\SonarScanner.MSBuild.PreProcessor.Tests\SonarScanner.MSBuild.PreProcessor.Tests.csproj", "{81E8D103-270F-4363-87D1-72BE7FC4F94A}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SonarScanner.Shim", "src\SonarScanner.Shim\SonarScanner.Shim.csproj", "{C28A09BF-6025-4EA1-90D7-4D5EC0AF815D}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SonarScanner.MSBuild.Shim", "src\SonarScanner.MSBuild.Shim\SonarScanner.MSBuild.Shim.csproj", "{C28A09BF-6025-4EA1-90D7-4D5EC0AF815D}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SonarScanner.Shim.Tests", "tests\SonarScanner.Shim.tests\SonarScanner.Shim.Tests.csproj", "{F43FC14B-5249-46F4-9E89-F4AAFDDB27ED}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SonarScanner.MSBuild.Shim.Tests", "tests\SonarScanner.MSBuild.Shim.Tests\SonarScanner.MSBuild.Shim.Tests.csproj", "{F43FC14B-5249-46F4-9E89-F4AAFDDB27ED}"
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SonarQube.Bootstrapper", "src\SonarQube.Bootstrapper\SonarQube.Bootstrapper.csproj", "{60AACBB4-1661-4EEB-A029-91289C1D3F7E}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SonarQube.Bootstrapper.Tests", "tests\SonarQube.Bootstrapper.tests\SonarQube.Bootstrapper.Tests.csproj", "{3410EA30-5F7A-4E0F-B5F6-8E83077A20DB}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SonarScanner.MSBuild.Tests", "tests\SonarScanner.MSBuild.Tests\SonarScanner.MSBuild.Tests.csproj", "{3410EA30-5F7A-4E0F-B5F6-8E83077A20DB}"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Packaging", "Packaging", "{37FB4C72-0F3D-4BF2-9F78-2780A76746A2}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SonarQube.TeamBuild.PostProcessor.Tests", "tests\SonarQube.TeamBuild.PostProcessor.tests\SonarQube.TeamBuild.PostProcessor.Tests.csproj", "{BF55995C-B05B-44D0-B80E-2366CC5ACCDE}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SonarScanner.MSBuild.PostProcessor.Tests", "tests\SonarScanner.MSBuild.PostProcessor.Tests\SonarScanner.MSBuild.PostProcessor.Tests.csproj", "{BF55995C-B05B-44D0-B80E-2366CC5ACCDE}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "BuildAgentPayload", "src\PackagingProjects\BuildAgentPayload.csproj", "{ECEADD3A-5992-420F-950D-510BBDDC06E1}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "BuildAgentPayload", "src\PackagingProjects\BuildAgentPayload.csproj", "{ECEADD3A-5992-420F-950D-510BBDDC06E1}"
```

Comment:
I think you should revert this change of GUID. As the GUID is used by the scanner (and SonarQube) to identify a module this will create a new module in SQ and delete/close the old one (closing all related issues and opening new ones).

Reply:
Well spotted!
The first Guid is a project type, the second one is the unique one, so it should be fine: https://stackoverflow.com/questions/2327202/what-are-the-project-guids-in-a-visual-studio-solution-file-used-for

But, VS is changing the Guids and it's a bug - the `FA..` one should be used:
https://github.com/dotnet/project-system/issues/1821

From comments in there, it seems like a bening change, but perhaps it's related to the build failing on the build machine?

Reply:
I don't know if that's the reason of the failure but this is an "evil bug"...

Reply:
The safest thing to do would be to explicitly put the _ProjectGuid_s into the new-style project files.
It won't make any difference to VS, but it does to us since those guids are effectively the unique project ids in _next_.